### DISCUSSION TITLE
Custom arrow in Select

### DISCUSSION THREAD

**Discussion by giantrobotbee** [ariakit/ariakit#3310] [Questions] [answered]
I'm trying to customize the arrow element appearing in a `Select` element, and I figured the easiest way was to do something like this, where I just use the `render` prop of `SelectArrow`:

```
      <SelectProvider defaultValue={defaultValue}>
        {label && <SelectLabel>{label}</SelectLabel>}
        <Select>
          <SelectArrow
            render={(props) => {
              return (
                <Image
                  alt="select arrow down"
                  width={20}
                  height={20}
                  src="icons/select-arrow-down.svg"
                />
              );
            }}
          />
        </Select>
        <SelectPopover gutter={8} sameWidth {...popoverOptions}>
          {placeholder && <SelectItem value="">{placeholder}</SelectItem>}
          {renderItems(selectItems)}
        </SelectPopover>
      </SelectProvider>
```

However, for some reason, when I do this, the button no longer renders the select item text.
![Screenshot 2024-01-08 at 15 11 00](https://github.com/ariakit/ariakit/assets/253703/dcf9f487-ccb8-49c0-8c53-71c3dc54f31d)
   
   Any idea how to fix this?

**Comment by diegohaz** [comment, marked as answer]
You don't need `SelectArrow`. Simply render `Image` directly. Also, you're replacing the `Select` children, so you have to render the value manually:

```jsx
const [value, setValue] = useState(defaultValue);

<SelectProvider value={value} setValue={setValue}>
  <Select>
    {value}
    <Image ... />
```

In the future, we'll introduce a `<SelectValue />` component so you can use it instead of managing the `value` state.
